### PR TITLE
[#210] Nation view changed (connects #210 )

### PR DESCRIPTION
**Pull request by Pr0chin**
look

### CODE REVIEW

File: src/app/nation/nation.component.html [hunk 1]
```diff
@@ -1,38 +1,45 @@
...
-        <button type="button" mat-raised-button color="primary" (click)="view(element.nationurl)" i18n>View</button>
-        <button mat-raised-button color="primary" (click)="communityList(element.name)" i18n> Communities </button>
-        <button mat-raised-button color="warn" (click)="deleteClick(element)" i18n> Delete </button>
-      </mat-cell>
-    </ng-container>
-    <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
-    <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
-  </mat-table>
-  <mat-paginator #paginator [pageSize]="10" [pageSizeOptions]="[5, 10, 20]" i18n></mat-paginator>
+    <mat-table #table [dataSource]="nations" matSort>
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: indentation

File: src/app/nation/nation.scss [hunk 2]
```diff
@@ -4,15 +4,58 @@
...
+  float: right;
+  margin: 10px;
+}
+.matSearch {
+  position: fixed;
+  margin-left: -35px;
+  float:left;
+}
+.matAdd {
+ width: 30px;
```
Comment (lmmrssa) [inline src/app/nation/nation.scss]: indentation

File: src/app/nation/nation.scss [hunk 3]
```diff
@@ -4,15 +4,58 @@
...
 }
 
-.example-full-width {
-  width: 100%;
+mat-table-layout {
+  margin: auto;
+}
+.add-nation {
+  min-height: 60px;
+  background-color: #2196f3;
```
Comment (lmmrssa) [inline src/app/nation/nation.scss]: color can go to shared file and used from there as its scss

File: src/app/nation/nation.scss [hunk 4]
```diff
@@ -4,15 +4,58 @@
...
+  text-align: center;
+  font-weight: bold;
+  color: #212121;
+}
+
+.mat-paginator {
+  float : left;
+}
+
+mat-form-field {
```
Comment (razu9861) [inline src/app/nation/nation.scss]: this scss will change all form-field if any form-filed is added in future

File: src/app/nation/nation.component.html [hunk 5]
```diff
@@ -1,38 +1,45 @@
...
 <div class="mat-elevation-z8">
-  <h3 class="header" i18n> Nation List </h3>
-  <button class="header" mat-raised-button color="primary" (click)="openNationAddForm()" data-toggle="modal" data-target="#nationAdd" i18n>Add</button>
-  <div class="search-header">
-    <mat-form-field>
-      <input matInput (keyup)="applyFilter($event.target.value)" i18n-placeholder placeholder="Search">
-    </mat-form-field>
+  <div class="add-nation">
+    <span class="labelNation">Your Current Lists of Nations</span>
+    <span class="addNation">Add a Nation <mat-icon class = "matAdd" (click)="openNationAddForm()">add</mat-icon></span>
```
Comment (paulbert) [inline src/app/nation/nation.component.html]: Please use a `mat-fab` here

File: src/app/nation/nation.scss [hunk 6]
```diff
@@ -4,15 +4,58 @@
...
+}
+
+.labelNation {
+  color: #FFFFFF;
+  position: fixed;
+  margin: 10px;
+}
+
+.addNation{
+  color: #FFFFFF;
```
Comment (paulbert) [inline src/app/nation/nation.scss]: Repetition with `color` & `margin` since these share a parent.

File: src/app/nation/nation.scss [hunk 7]
```diff
@@ -4,15 +4,58 @@
...
+  margin: auto;
+}
+.add-nation {
+  min-height: 60px;
+  background-color: #2196f3;
+}
+
+.labelNation {
+  color: #FFFFFF;
+  position: fixed;
```
Comment (paulbert) [inline src/app/nation/nation.scss]: Please use `display: inline-block`.  `position: fixed` is intended for a different purpose.

**Comment by paulbert** [comment]
This PR was relying too much on custom css for the nation component when simple Material elements could be used.  A few notes:

* We should not have so many classes that are specific to components.  I deleted the `nation.scss` file entirely and with a few additions to the main `style.scss` we have a similar layout that doesn't break with smaller screens.  We should simplify our CSS and using mainly Material elements will help with that.
* Another point about simplification: we don't have specific designs to work off of, so tweaking the margin from `-59px` to `-68px` ([see line 23](https://github.com/ole-vi/planet/pull/212/commits/98e742bc608c957630ce8fcd64a798d14ff495de#diff-5cd4d09dfd9fb7ad43eda76c346991deL23)) is not a good use of our time.  We can tweak when UX has designs for us.
* Do not create classes that modify built in Material classes.  We want to make sure there is consistent display for the different Material elements or a clear custom class that we have written that adjusts the element.
* Colors should always refer to colors from the Material themes we use.  Either use `color="primary"` or now I added some helper classes so `class="primary-color"` will work.  I thought we were clear about this last night, but I now see that the `color="warn"` for the delete button was changed to a specific class with a hard coded color in the last commit.

I'm going to add another PR for the theme colors.  Please do not update this PR to overwrite the color selections.

**Comment by Pr0chin** [comment]
@paulbert ok :+1:

File: src/app/nation/nation.component.html [hunk 8]
```diff
@@ -1,38 +1,48 @@
...
-    <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
-  </mat-table>
-  <mat-paginator #paginator [pageSize]="10" [pageSizeOptions]="[5, 10, 20]" i18n></mat-paginator>
-</div>
+  </mat-toolbar-row>
+
+  <mat-toolbar-row class="primary-color font-size-1">
+    <span>Your Current Lists of Nations</span>
+    <span class="toolbar-fill"></span>
+    <span>Add a Nation</span>
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: translation

File: src/app/nation/nation.component.html [hunk 9]
```diff
@@ -1,38 +1,48 @@
...
-    </ng-container>
-    <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
-    <mat-row *matRowDef="let row; columns: displayedColumns;"></mat-row>
-  </mat-table>
-  <mat-paginator #paginator [pageSize]="10" [pageSizeOptions]="[5, 10, 20]" i18n></mat-paginator>
-</div>
+  </mat-toolbar-row>
+
+  <mat-toolbar-row class="primary-color font-size-1">
+    <span>Your Current Lists of Nations</span>
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: translation

File: src/app/nation/nation.component.html [hunk 10]
```diff
@@ -1,38 +1,48 @@
...
+    <mat-header-cell *matHeaderCellDef mat-sort-header i18n>Admin Name</mat-header-cell>
+    <mat-cell *matCellDef="let element">{{element.admin_name}}</mat-cell>
+  </ng-container>
+  <!-- URL Column -->
+  <ng-container matColumnDef="nationurl">
+    <mat-header-cell *matHeaderCellDef mat-sort-header i18n>Nation URL</mat-header-cell>
+    <mat-cell *matCellDef="let element">{{element.nationurl}}</mat-cell>
+  </ng-container>
+    <!-- Action Column -->
+    <ng-container matColumnDef="action">
```
Comment (lmmrssa) [inline src/app/nation/nation.component.html]: indentation

Comment (nirojdyola) [reply src/app/nation/nation.component.html]: Fix indentation

**Pull request merged**